arch/lkl/kernel/misc.c

```diff
@@ -1,5 +1,23 @@
 #include <linux/seq_file.h>
 
+// Functions __generic_xchg_called_with_bad_pointer and wrong_size_cmpxchg
+// are called in __generic_xchg and __generic_cmpxchg_local respectively.
+// They should be optimized out by the compiler due to the function
+// inlining. However, when building with clang there are some instances
+// where the functions aren't inlined and, thus, the compile-time optimization
+// doesn't eliminate them entirely. As a result, the linker throws
+// unresololved symbols error. The fix below define these functions to
+// bypass the link-time error.
+void __generic_xchg_called_with_bad_pointer(void)
```

Comment:
I also saw this issue when I build LKL w/ modified optimization options (e.g., `CFLAGS_page_alloc.o=-O0` in mm/Makefile IIRC).  I couldn't find a proper fix for that then work-around with the similar fix as this.

I'm fine for this fix with the detailed comment.  maybe add TODO tag ?

Reply:
Added TODO tag as suggested.